gstd/gstd_http.c

```diff
@@ -0,0 +1,436 @@
+  GstdReturnCode ret = GSTD_EOK;
+
+  g_return_val_if_fail (server, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (msg, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (session, GSTD_NULL_ARGUMENT);
+
+  address = soup_message_get_uri (msg);
+
+  if (!name) {
+    ret = GSTD_BAD_VALUE;
+    GST_ERROR_OBJECT (session,
+        "Wrong query param provided, \"name\" doesn't exist");
+    goto out;
+  }
+  if (!description) {
+    ret = GSTD_BAD_VALUE;
+    GST_ERROR_OBJECT (session,
+        "Wrong query param provided, \"description\" doesn't exist");
+    goto out;
+  }
+
+  message = g_strdup_printf
+      ("create %s %s %s", soup_uri_get_path (address), name, description);
+  ret = gstd_parser_parse_cmd (session, message, output);
+  g_free (message);
+
+out:
+  return ret;
+}
+
+static GstdReturnCode
+do_put (SoupServer * server, SoupMessage * msg, char *name, char **output,
+    GstdSession * session)
+{
+  gchar *message = NULL;
+  SoupURI *address = NULL;
+  GstdReturnCode ret = GSTD_EOK;
+
+  g_return_val_if_fail (server, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (msg, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (session, GSTD_NULL_ARGUMENT);
+
+  address = soup_message_get_uri (msg);
+
+  if (!name) {
+    ret = GSTD_BAD_VALUE;
+    GST_ERROR_OBJECT (session,
+        "Wrong query param provided, \"name\" doesn't exist");
+    goto out;
+  }
+
+  message = g_strdup_printf ("update %s %s", soup_uri_get_path (address), name);
+  ret = gstd_parser_parse_cmd (session, message, output);
+  g_free (message);
+
+out:
+  return ret;
+}
+
+static GstdReturnCode
+do_delete (SoupServer * server, SoupMessage * msg, char *name,
+    char **output, GstdSession * session)
+{
+  gchar *message = NULL;
+  SoupURI *address = NULL;
+  GstdReturnCode ret = GSTD_EOK;
+
+  g_return_val_if_fail (server, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (msg, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (session, GSTD_NULL_ARGUMENT);
+
+  address = soup_message_get_uri (msg);
+
+  if (!name) {
+    ret = GSTD_BAD_VALUE;
+    GST_ERROR_OBJECT (session,
+        "Wrong query param provided, \"name\" doesn't exist");
+    goto out;
+  }
+
+  message = g_strdup_printf ("delete %s %s", soup_uri_get_path (address), name);
+  ret = gstd_parser_parse_cmd (session, message, output);
+  g_free (message);
+
+out:
+  return ret;
+}
+
+static void
+do_request (SoupServer * server, SoupMessage * msg, GHashTable * query,
+    GstdSession * session, GstdHttpVerbs request_verb)
+{
+  gchar *response = NULL;
+  gchar *name = NULL;
+  gchar *description_pipe = NULL;
+  SoupURI *address = NULL;
+  GstdReturnCode ret = GSTD_EOK;
+  gchar *output = NULL;
+  const gchar *description = NULL;
+  SoupStatus status = SOUP_STATUS_OK;
+  const gchar *query_text = NULL;
+
+  g_return_if_fail (server);
+  g_return_if_fail (msg);
+  g_return_if_fail (session);
+
+  address = soup_message_get_uri (msg);
+
+  query_text = soup_uri_get_query (address);
+
+  if (!(msg->method == SOUP_METHOD_GET)) {
```

Comment:
I don't like this here. I think it couples too much to the chosen verbs. I'd only search in the hash table if it's available. Then let the specific verbs decide if they need it or not.